### DISCUSSION TITLE
make pass state part of this repo's structure

### DISCUSSION THREAD

**Discussion by blaggacao** [nrdxp/nixdev#187] [Ideas]
**Is your feature request related to a problem? Please describe.**
My passwords are part of my environment, I want them to be persisted under the vigorous structure and management of this repository, instead of acquiring a satellite repository state.

**Describe the solution you'd like**
Thanks to the advent and example of agenix, it should be at least conceivable to implement a pass-compliant script that honours the agenix precedent of stowing configuration metadata.

**Describe alternatives you've considered**
This would be so awesome that no alternative comes even close :smiley: 

**Additional context**
- https://github.com/ryantm/agenix
- https://github.com/ryantm/agenix/pull/14 (agenix itself might be even sufficiently extensible or something called `agenix-pass` to keep provenance proof in the name)

**Comment by nrdxp** [comment]
Indeed, I'm glad you opened this, as I have been thinking about this alot. I'll be pretty busy the next couple of days, but I should have some time once the new year starts to take a good crack at this.

**Reply by blaggacao** [reply]
I think https://github.com/divnix/devos/discussions/180#discussioncomment-497534 inadvertedly brought the solution (`impermanence`)
